staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -291,7 +291,8 @@ func (le *LeaderElector) release() bool {
 		return true
 	}
 	leaderElectionRecord := rl.LeaderElectionRecord{
-		LeaderTransitions: le.observedRecord.LeaderTransitions,
+		LeaderTransitions:    le.observedRecord.LeaderTransitions,
```

Comment:
Ideally it seems like we should use something like Server Side Apply to only change the LeaderTransitions fields. 

I may be missing something, but the use case for this code seems to be we have lost the lease. We (eg. identity baz) were the leader but lost the lease (we no longer hold the lock) and are trying to record a bump in the number of transitions. Recoding the bump in transitions is failing because we don't allow a lease to have a duration of 0 and we are setting it to zero because that is the default value for that field. Your fixing that by using the duration set by the current owner (not us, example holder identity zat) to the new record we are writing. It seems like since we didn't explicitly set the holder identity on the update, so we will be over writing it with the default for the field (empty string). 

Can we add a check for holder identity after the release, in your test to make sure that is not happening?